src/xrNetServer/NET_PlayersMonitor.h

```diff
@@ -34,7 +34,7 @@ class PlayersMonitor
     {
         if (now_iterating_in_net_players || now_iterating_in_net_players_disconn)
         {
-            if (iterator_thread_id == Threading::GetCurrThreadId())
+            if (iterator_thread_id == reinterpret_cast<size_t>(Threading::GetCurrThreadId()))
```

Comment:
The pthread manpage advises that `pthread_equal` should be used to compare `pthread_t` structs, because the details of what that struct entails are implementation-defined & not guaranteed to be an arithmetic type.

```suggestion
#ifndef XR_PLATFORM_WINDOWS
            if (iterator_thread_id == Threading::GetCurrThreadId())
#else
            if (pthread_equal(iterator_thread_id, Threading::GetCurrThreadId()))
#endif
```
\+ some more initialisation code for `iterator_thread_id` to not assume an arithmetic type. With that all the other `reinterpret_cast<size_t>(Threading::GetCurrThreadId());` should be dropped as well.

Reply:
Thanks, fixed!